sonic_data_client/db_client.go

```diff
@@ -367,52 +368,98 @@ func ValToResp(val Value) (*gnmipb.SubscribeResponse, error) {
 	}
 }
 
-func GetTableKeySeparator(target string) (string, error) {
+func GetTableKeySeparator(target string, ns string) (string, error) {
 	_, ok := spb.Target_value[target]
 	if !ok {
 		log.V(1).Infof(" %v not a valid path target", target)
 		return "", fmt.Errorf("%v not a valid path target", target)
 	}
 
-	var separator string = sdcfg.GetDbSeparator(target)
+	var separator string = sdcfg.GetDbSeparator(target, ns)
 	return separator, nil
 }
 
+func GetRedisClientsForDb(target string) map[string]*redis.Client {
+	redis_client_map := make(map[string]*redis.Client)
+	if sdcfg.CheckDbMultiNamespace() {
+		ns_list, _ := sdcfg.GetDbNonDefaultNamespaces()
+		for _, ns := range ns_list {
+			redis_client_map[ns] = Target2RedisDb[ns][target]
+		}
+	} else {
+		ns := sdcfg.GetDbDefaultNamespace()
+		redis_client_map[ns] = Target2RedisDb[ns][target]
+	}
+	return redis_client_map
+}
+
+// This function get target present in GNMI Request and
+// returns: 1. DbName (string) 2. Is DbName valid (bool)
+//          3. DbNamespace (string) 4. Is DbNamespace present in Target (bool)
+func IsTargetDb(target string) (string, bool, string, bool) {
+	targetname := strings.Split(target, "/")
+	dbName := targetname[0]
+	dbNameSpaceExist := false
+	dbNamespace := sdcfg.GetDbDefaultNamespace()
+
+	if len(targetname) > 2 {
+		log.V(1).Infof("target format is not correct")
+		return dbName, false, dbNamespace, dbNameSpaceExist
+	}
+
+	if len(targetname) > 1 {
+		dbNamespace = targetname[1]
+		dbNameSpaceExist = true
+	}
+	for name, _ := range spb.Target_value {
+		if name == dbName {
+			return dbName, true, dbNamespace, dbNameSpaceExist
+		}
+	}
+
+	return dbName, false, dbNamespace, dbNameSpaceExist
+}
+
 // For testing only
 func useRedisTcpClient() {
-	for dbName, dbn := range spb.Target_value {
-		if dbName != "OTHERS" {
-			// DB connector for direct redis operation
-			var redisDb *redis.Client
-			if UseRedisLocalTcpPort {
-				redisDb = redis.NewClient(&redis.Options{
-					Network:     "tcp",
-					Addr:        sdcfg.GetDbTcpAddr(dbName),
-					Password:    "", // no password set
-					DB:          int(dbn),
-					DialTimeout: 0,
-				})
+	if UseRedisLocalTcpPort {
```

Comment:
1) Code nesting could be reduced.
```
if !UseRedisLocalTcpPort {
    return
}
```
2) Should `else` case be handled to?

Reply:
updated for 1st point. For 2nd point  this function used only from perspective for unit-test. `else` is not needed if we somehow reach that condition  unit-test will fail.